src/sentry/integrations/webhook.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any
+
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint
+
+logger = logging.getLogger(__name__)
+
+
+class IntegrationWebhookEndpoint(Endpoint, ABC):
+    owner = ApiOwner.ECOSYSTEM
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    authentication_classes = ()
+    permission_classes = ()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.log_extra = {}
+
+    @property
+    @abstractmethod
+    def provider(self) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def unpack_payload(self, request: Request, **kwargs) -> Any:
+        raise NotImplementedError
+
+    @abstractmethod
+    def authenticate(self, request: Request, **kwargs) -> Any:
+        raise NotImplementedError
+
+    @csrf_exempt
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        self.log_extra.update(
+            {"request_method": self.request.method, "request_path": self.request.path}
+        )
```

Comment:
Should this be before the dispatch call? Otherwise, these fields may not be populated in the Jira base webhook logger if I'm following the code correctly.

Reply:
`self.request` doesn't exist until `super().dispatch()` is called

https://github.com/getsentry/sentry/blob/6670e30f397d5793e4ba91526e4fc915ecbbcbf8/src/sentry/api/base.py#L366-L381

Reply:
Ahh that's not great. Doesn't this also mean we have a chicken-egg problem? We are removing the `request_method` and `request_path` params from this log here, in favor of the new member variable: https://github.com/getsentry/sentry/blob/cc43759f7e8512cc8a3fbbc03fb7dad712566813/src/sentry/integrations/jira/webhooks/base.py#L60

But this is called by the parent dispatch method here: https://github.com/getsentry/sentry/blob/eb8d0c6dd1f47f4d74998bd2bbc357c72577cfbe/src/sentry/api/base.py#L438

So wouldn't this log always be missing these values?